app/components/select-percentage.tsx

```diff
@@ -0,0 +1,183 @@
+                  >
+                    {label}
+                  </label>
+                </div>
+              ))}
+            </div>
+          </fieldset>
+
+          <div className="w-full flex flex-row items-center justify-start space-x-2">
+            <img
+              src="/time-half-pass.svg"
+              alt="time half pass"
+              className="w-auto h-auto"
+            />
+            <span className="text-[13px] text-[#F9F9F9] text-[400]">
+              Base Token
+            </span>
+          </div>
+
+          <div className="w-full flex gap-8 sm:gap-16">
+            {[
+              {
+                logo: "/usdt-logo.svg",
+                label: "Usdt",
+                value: "usdt",
+                id: "usdt",
+              },
+              {
+                logo: "/usdc-logo.svg",
+                label: "Usdc",
+                value: "usdc",
+                id: "usdc",
+              },
+            ].map(({ logo, label, value, id }) => (
+              <div
+                key={id}
+                className={`w-full flex items-center justify-center mx-auto`}
+              >
+                <input
+                  type="radio"
+                  id={id}
+                  name="radio-group"
+                  value={value}
+                  className={`hidden`}
+                  onClick={() => handleBaseTokenChange(value)}
+                />
+                <label
+                  htmlFor={id}
+                  className={`w-full relative flex flex-row space-x-2 items-center justify-start cursor-pointer select-none`}
+                >
+                  <img
+                    src={logo}
+                    alt={label}
+                    className="w-[48px] h-[48px] sm:w-[60px] sm:h-[60px] rounded-full"
+                  />
+                  <div className="w-full flex flex-col space-y-0 items-start justify-start">
+                    <span className="block w-full text-[#F9F9F9] text-[14px] sm:text-[16px] font-[400]">
+                      {label}
+                    </span>
+                    <span className="block w-full text-[#433B5A] text-[12px] sm:text-[14px] font-[600]">
+                      {label.toUpperCase()}
+                    </span>
+                  </div>
+                  {baseToken == value ? (
+                    <img
+                      src="/blue-checkmark-circle.svg"
+                      alt="check"
+                      className="w-[16px] h-[16px] sm:w-[20px] sm:h-[20px] absolute right-0"
+                    />
+                  ) : (
+                    ""
+                  )}
+                </label>
+              </div>
+            ))}
+          </div>
+
+          <div className="w-full flex flex-none items-center justify-center my-4 mx-auto bg-[#170F2E] h-[1px]"></div>
+
+          <div className="w-full mx-auto flex items-center justify-center">
+            <button
```

Comment:
add a disabled prop and styles. the button should be disabled when, there is neither an inputted amount nor a selected percentage. it should also be disabled when a base token is not selected.